specs/liberty/api.rst

```diff
@@ -0,0 +1,280 @@
+
+    {
+      "id": "f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+      "name": "Main Website",
+      "type": "neutron_lbaas_v2",
+      "region": "RegionTwo",
+      "neutron_id": "df40954c-982a-4778-a1b2-32b93cf75af9",
+      "status": "ACTIVE",
+      "endpoints": [
+        "10.10.0.1"
+      ]
+    }
+
+
+Pool Member JSON Snippet - Neutron Port
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id": "f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+      "name": "Main Website",
+      "type": "neutron_port",
+      "region": "RegionTwo",
+      "neutron_id": "78fa9c7f-f200-49df-912b-ae4679fd21e9",
+      "status": "ACTIVE",
+      "endpoints": [
+        "10.10.0.1"
+      ]
+    }
+
+Pool Member JSON Snippet - IP
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id": "f9fc6210-8efa-4492-9b4f-d87ba98f35b8",
+      "name": "Main Website",
+      "type": "IP",
+      "status": "ACTIVE",
+      "endpoints": [
+        "10.10.0.1"
+      ]
+    }
+
+
+Monitor JSON Snippet - TCP Basic
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id" : "f9fc6210-8efa-4492-9b4f-d87ba98f35b8"
+      "type" : "TCP",
+      "delay" : 20,
+      "timeout": 10,
+      "max_retries": 3,
+      "port" : 22,
+      "name": "ssh_generic"
+    }
+ 
+Monitor JSON Snippet - Ping Basic
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id" : "f9fc6210-8efa-4492-9b4f-d87ba98f35b8"
+      "type" : "PING",
+      "delay" : 20,
+      "timeout": 10,
+      "max_retries": 3,
+      "name": "ping_generic"
+    }
+
+Monitor JSON Snippet - HTTP
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id" : "f9fc6210-8efa-4492-9b4f-d87ba98f35b8"
+      "delay": 20,
+      "timeout": 10,
+      "max_retries": 3,
+      "type": 'HTTP',
+      "http_method": 'GET',
+      "url_path": "/healthchecks",
+      "receive_string" "SUCCESS",
+      "expected_codes": [
+        200,
+        202
+      ],
+      "name": "http-generic",
+      "port" : 80
+    }
+
+Monitor JSON Snippet - HTTPS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id" : "f9fc6210-8efa-4492-9b4f-d87ba98f35b8"
+      "delay": 20,
+      "timeout": 10,
+      "max_retries": 3,
+      "type": 'HTTPS',
+      "insecure": False,
+      "http_method": 'POST',
+      "payload": {
+        "content_type" : "application/json",
+        "content" "{'demo': 'JSON', 'Blob': 'of data'}"
+        }
+      },
+      "ca_cert": "<CA_CERT_BLOB>"
+      "url_path": "/healthchecks",
+      "receive_string" "SUCCESS",
+      "expected_codes": [
+        200,
+        202
+      ],
+      "name": "https-generic",
+      "port" : 443
+    }
+
+Endpoints
+=========
+
+All of these endpoints will have the usual CRUD methods available
+
+/v0.1/gslbs
+-----------
+
+Returns a list of GLSBs
+
+/v0.1/gslbs/<uuid>
+------------------
+
+Returns a GLSB
+
+/v0.1/gslbs/<uuid>/pools
+------------------------
```

Comment:
We might also need a way for us to associate pools to load balancers either via separate API or as part the load balancer create API included in the load balancer JSON snippet.